invenio_users_resources/records/api.py

```diff
@@ -8,62 +8,111 @@
-        "notifications",
-        {
-            "enabled": True,
-        },
-    )
+from .models import DomainAggregateModel, GroupAggregateModel, UserAggregateModel
+from .systemfields import (
+    AccountStatusField,
+    AccountVisibilityField,
+    DomainCategoryNameField,
+    DomainField,
+    DomainOrgField,
+    DomainStatusNameField,
+    IsNotNoneField,
+    UserIdentitiesField,
+)
+
+EmulatedPID = namedtuple("EmulatedPID", ["pid_value"])
+"""Emulated PID"""
+
+
+class AggregatePID:
+    """Helper emulate a PID field."""
+
+    def __init__(self, pid_field):
+        """Constructor."""
+        self._pid_field = pid_field
+
+    def __get__(self, record, owner=None):
+        """Evaluate the property."""
+        if record is None:
+            return GetRecordResolver(owner)
+        return EmulatedPID(record[self._pid_field])
+
+
+class GetRecordResolver(object):
+    """Resolver that simply uses get record."""
+
+    def __init__(self, record_cls):
+        """Initialize resolver."""
+        self._record_cls = record_cls
+
+    def resolve(self, pid_value):
+        """Simply get the record."""
+        return self._record_cls.get_record(pid_value)
+
+
+class BaseAggregate(Record):
+    """An aggregate of information about a user group/role."""
+
+    metadata = None
+    """Disabled metadata field from the base class."""
+
+    def __getitem__(self, name):
+        """Get a dict key item."""
+        try:
+            return getattr(self.model, name)
+        except AttributeError:
+            raise KeyError(name)
+
+    def __repr__(self):
+        """Create string representation."""
+        return f"<{self.__class__.__name__}: {self.model.data}>"
 
-    return data
+    def __unicode__(self):
+        """Create string representation."""
+        return self.__repr__()
 
+    @classmethod
+    def from_model(cls, sa_model):
+        """Create an aggregate from an SQL Alchemy model."""
+        return cls({}, model=cls.model_cls(model_obj=sa_model))
+
+    def _validate(self, *args, **kwargs):
+        """Skip the validation."""
+        pass
 
-def parse_role_data(role):
-    """Parse the role's information into a dictionary."""
-    data = {
-        "id": role.id,
-        "name": role.name,
-        "description": role.description,
-        "is_managed": role.is_managed,
-    }
-    return data
+    def commit(self):
+        """Update the aggregate data on commit."""
+        # You can only commit if you have an underlying model object.
+        if self.model._model_obj is None:
+            raise ValueError(f"{self.__class__.__name__} not backed by a model.")
+        if self.model._model_obj not in db.session:
+            with db.session.begin_nested():
+                # make sure we get an id assigned
+                db.session.add(self.model._model_obj)
+        # Basically re-parses the model object.
+        model = self.model_cls(model_obj=self.model._model_obj)
+        self.model = model
```

Comment:
Wouldn't `db.session.flush()`cut it here to reparse the obj?
```suggestion
        db.session.flush()
```

Reply:
@Samk13 Thanks for the review on this and the other PRs :-) In this case it’s simply to ensure that the aggregate model pick up changes that was applied to the User/Domain model from Invenio-Accounts. E..g when you apply datastore.activate_user() it changes the underlying model object outside of the aggregate API, and when the resource has to serialise back out the values to the REST API it uses an stale cache. Flush would send changes to the DB, but that’s not needed at this point yet.

Reply:
Got it, no flush() needed, prioritizing performance. Thanks for the clarification.